app/src/org/commcare/logging/DataChangeLogger.kt

```diff
@@ -0,0 +1,156 @@
+package org.commcare.logging
+
+import android.content.Context
+import android.os.Environment
+import android.util.Log
+import com.crashlytics.android.Crashlytics
+import org.apache.commons.lang3.StringUtils
+import org.commcare.android.logging.ReportingUtils
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.FileUtil
+import org.javarosa.core.io.StreamsUtil
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ *  Used to log some crtitical platform events like DB migrations, Clear user data,
+ *  app installations etc. All these logs are backed by file storage and are accessible
+ *  from outside CommCare
+ */
+class DataChangeLogger {
```

Comment:
100% for my learning/understanding: Can you explain the reasoning behind the general structure you used for this class, in terms of having a non-static init methods and then the `companion object` that houses everything else? From what I read, anything placed inside the `companion object` is basically equivalent to being a static method; is that right? If so, I'm still a little uncertain as to why the class fields go in there as well, and the reasoning behind keeping certain methods outside of that.

Reply:
Since I wanted all constants and both file accessors to be static, I have put them in `companion` . Though I get your confusion for not making `init` static which I see no harm in. I think since we only ever want one instance of `DataChangeLogger` we can instead make this whole class a Kotlin object which will also remove the need for the `companion`.

Reply:
Got it ok yea, obviously I'm super unfamiliar with Kotlin coding conventions, but that does sound a little more intuitive to me than the current setup.